### ISSUE TITLE
"az k8s-configuration flux create" leads to k8s service account "flux-applier" requesting cluster-role-binding instead of role-binding

### ISSUE THREAD

**Issue by gstolz** [Azure/azure-cli#22815]
**Related command**
`az k8s-configuration flux create --resource-group test --cluster-name test-aks --name cluster-config --namespace cluster-config --cluster-type managedClusters --scope namespace --url https://github.com/Azure/gitops-flux2-kustomize-helm-mt --branch main --kustomization name=infra path=./infrastructure prune=true --kustomization name=apps path=./apps/staging prune=true`

**Describe the bug**
I am following the [tutorial for Flux v2 on AKS](https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/tutorial-use-gitops-flux2)
The error occurs independently of Azure portal or az cli.
So setting up Flux on Azure creates a Git repo and two Kustomization config. The git repo is fine, the kustomize configs cannot reconcile, because of a permission issue:
`Namespace/podinfo dry-run failed, reason: Forbidden, error: namespaces podinfo is forbidden: User system:serviceaccount:cluster-config:flux-applier cannot patch resource namespaces in API group  in the namespace podinfo`
So basically the service principle needs higher permission  (cluster-role) to make changes on the cluster, even-though the scope was set to "namespace". This is quite important since we would like to set this up in a multi-tenant environment.

**To Reproduce**
Follow the [tutorial](https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/tutorial-use-gitops-flux2) until the above mentioned command.

**Expected behavior**
Expectation would be that the service principal's actions are limited to the namespace it was set to.

**Environment summary**
Using az cli 2.37 (az extensions all updated to latest version) and AKS/K8S 1.23.5

**Additional context**

**Comment by yonzhan** [comment]
route to CXP team

**Comment by gstolz** [comment]
Hi @yonzhan , is there already some feedback from "CXP" team?

**Comment by navba-MSFT** [comment]
@gstolz Apologies for the late reply. Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@gstolz You can use [Pamir/flux2-kustomize-helm-example: A GitOps workflow example for multi-env deployments with Flux, Kustomize and Helm. (github.com)](https://github.com/Pamir/flux2-kustomize-helm-example)

 Please let me know if you see any issues with this. Awaiting your reply.

**Comment by navba-MSFT** [comment]
@gstolz I wanted to do quick follow-up to check if you had a chance to follow the above article and check if that helps. Awaiting your reply.

**Comment by navba-MSFT** [comment]
@gstolz The action is currently pending on you to test the above plan. We will now proceed with closure of this GitHub issue. If you need any further assistance on this issue in future, please feel free to reopen this thread. We would be happy to help.

**Comment by jasric89** [comment]
Hi, I have had a similar error when passing, kustomization flags I'm met with yaml errors and the command does not know what to do. Also with FluxV2 if you can't use the Azure extension and you try and build flux manually like in the tutorial you link to. Flux will fail and not read the repo and it cant apply kustomizations properly. 

Here is my long error log: 

`az : DEBUG: cli.knack.cli: Command arguments: ['k8s-configuration', 'flux', 'create', '--resource-group', 'ncba-core-test-aks-rg', '--cluster-name', 'ncba-core-test-k8s', '--cluster-type', 'connectedClusters', '--name', 'fluxv2', '--scope', 
'cluster', '--namespace', 'flux-system', '--kind', 'git', '--url', 'https://<USERNAME>:<PATKEY>@dev.azure.com/projectname', '--branch', 'main', '--kustomization', 
'name=infra', 'path=./infrastructure/test', 'sync_interval=120', 'prune=true', 'force=true', '--kustomization', 'name=apps', 'path=./apps/test', 'depends_on=infra', 'sync_interval=120', 'prune=true', 'force=true', '--debug']
At line:1 char:1
+ az k8s-configuration flux create --resource-group ncba-core-test-aks- ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (DEBUG: cli.knac...ue', '--debug']:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
DEBUG: cli.knack.cli: __init__ debug log:
Cannot enable color.
DEBUG: cli.knack.cli: Event: Cli.PreExecute []
DEBUG: cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x018DB340>, <function OutputProducer.on_global_arguments at 0x01AA8AD8>, <function CLIQuery.on_global_arguments at 0x01AC7730>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
DEBUG: cli.azure.cli.core: Modules found from index for 'k8s-configuration': ['azext_k8s_configuration']
DEBUG: cli.azure.cli.core: Loading command modules:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands
DEBUG: cli.azure.cli.core: Total (0)                 0.000         0         0
DEBUG: cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_ai_examples', 'azext_next']
DEBUG: cli.azure.cli.core: Loading extensions:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
DEBUG: cli.azure.cli.core: k8s-configuration         0.006         4        16  C:\Users\jason\.azure\cliextensions\k8s-configuration
DEBUG: cli.azure.cli.core: Total (1)                 0.006         4        16  
DEBUG: cli.azure.cli.core: Loaded 4 groups, 16 commands.
DEBUG: cli.azure.cli.core: Found a match in the command table.
DEBUG: cli.azure.cli.core: Raw command  : k8s-configuration flux create
DEBUG: cli.azure.cli.core: Command table: k8s-configuration flux create
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x03F77B68>]
DEBUG: cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to 'C:\Users\jason\.azure\commands\2023-02-08.15-10-12.k8s-configuration_flux_create.10944.log'.
INFO: az_command_data_logger: command args: k8s-configuration flux create --resource-group {} --cluster-name {} --cluster-type {} --name {} --scope {} --namespace {} --kind {} --url {} --branch {} --kustomization {} {} {} {} {} --kustomization 
{} {} {} {} {} {} --debug
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x03F9ABB0>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x03FB0190>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x03FB0220>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
DEBUG: cli.azure.cli.core.util: azure.cli.core.util.handle_exception is called with an exception:
DEBUG: cli.azure.cli.core.util: Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 588, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/parser.py", line 261, in parse_args
  File "argparse.py", line 1826, in parse_args
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/parser.py", line 281, in parse_known_args
  File "argparse.py", line 1859, in parse_known_args
  File "argparse.py", line 2050, in _parse_known_args
  File "argparse.py", line 2027, in consume_positionals
  File "argparse.py", line 1936, in take_action
  File "argparse.py", line 1214, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/parser.py", line 281, in parse_known_args
  File "argparse.py", line 1859, in parse_known_args
  File "argparse.py", line 2050, in _parse_known_args
  File "argparse.py", line 2027, in consume_positionals
  File "argparse.py", line 1936, in take_action
  File "argparse.py", line 1214, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/parser.py", line 281, in parse_known_args
  File "argparse.py", line 1859, in parse_known_args
  File "argparse.py", line 2050, in _parse_known_args
  File "argparse.py", line 2027, in consume_positionals
  File "argparse.py", line 1936, in take_action
  File "argparse.py", line 1214, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/parser.py", line 281, in parse_known_args
  File "argparse.py", line 1859, in parse_known_args
  File "argparse.py", line 2068, in _parse_known_args
  File "argparse.py", line 2008, in consume_optional
  File "argparse.py", line 1936, in take_action
  File "C:\Users\jason\.azure\cliextensions\k8s-configuration\azext_k8s_configuration\action.py", line 39, in __call__
    validate_kustomization(values)
  File "C:\Users\jason\.azure\cliextensions\k8s-configuration\azext_k8s_configuration\validators.py", line 60, in validate_kustomization
    validate_duration("sync-interval", value)
  File "C:\Users\jason\.azure\cliextensions\k8s-configuration\azext_k8s_configuration\validators.py", line 147, in validate_duration
    raise InvalidArgumentValueError(
azure.cli.core.azclierror.InvalidArgumentValueError: Error! Invalid sync-interval.

ERROR: cli.azure.cli.core.azclierror: Error! Invalid sync-interval.
ERROR: az_command_data_logger: Error! Invalid sync-interval.
Specify a valid Go duration and try again (i.e. 1h2m5s)
DEBUG: cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x03F77C88>]
INFO: az_command_data_logger: exit code: 1
INFO: cli.__main__: Command ran in 0.863 seconds (init: 0.427, invoke: 0.436)
INFO: telemetry.main: Begin splitting cli events and extra events, total events: 1
INFO: telemetry.client: Accumulated 0 events. Flush the clients.
INFO: telemetry.main: Finish splitting cli events and extra events, cli events: 1
INFO: telemetry.save: Save telemetry record of length 3292 in cache
WARNING: telemetry.check: Negative: The C:\Users\jason\.azure\telemetry.txt was modified at 2023-02-08 15:01:11.203409, which in less than 600.000000 s
`

Here is the command I am entering: `
az k8s-configuration flux create --resource-group ncba-core-test-aks-rg --cluster-name ncba-core-test-k8s --cluster-type connectedClusters --name fluxv2 --scope cluster --namespace flux-system --kind git --url "https://<USERNAME>:<PATKEY>@dev.azure.com/projectname" --branch main --kustomization name=infra path=./infrastructure/test sync_interval=120 prune=true force=true --kustomization name=apps path=./apps/test depends_on=infra sync_interval=120 prune=true force=true --debug 2>&1 > fluxdebug.txt`
